ui/src/shared/components/TimeMachine/TimeMachine.tsx

```diff
@@ -261,6 +337,63 @@ class TimeMachine extends PureComponent<Props, State> {
     }))
   }
 
+  private get isFluxSource(): boolean {
+    // TODO: Update once flux is no longer a separate service
+    const {selectedService} = this.state
+    if (selectedService) {
+      return true
+    }
+    return false
+  }
+
+  private get fluxBuilder(): JSX.Element {
+    const {suggestions, body, status} = this.state
+    const {script, notify} = this.props
+
+    return (
+      <FluxContext.Provider value={this.getContext}>
+        <KeyboardShortcuts onControlEnter={this.getTimeSeries}>
+          <FluxQueryBuilder
+            body={body}
+            script={script}
+            status={status}
+            notify={notify}
+            context={this.getContext}
```

Comment:
why are you passing context in as both a context provider and a prop?

Reply:
good point